console/console-init/ui/src/modules/iot-device/dialogs/CreateDevice/CreateDevicePage.tsx

```diff
@@ -0,0 +1,266 @@
+        <BreadcrumbItem>
+          <Link id="cdetail-link-home" to={"/"}>
+            Home
+          </Link>
+        </BreadcrumbItem>
+        <BreadcrumbItem isActive={true}>{projectname}</BreadcrumbItem>
+      </Breadcrumb>
+    ),
+    [projectname]
+  );
+
+  useBreadcrumb(breadcrumb);
+
+  const getGateways = (gateways: string[]) => {
+    setAddedGateways(gateways);
+  };
+
+  const onCloseDialog = () => {
+    history.push(deviceListRouteLink);
+  };
+
+  const addGateway = {
+    name: "Add gateways",
+    component: <AddGateways returnGateways={getGateways} />
+  };
+
+  const AddCredentialWrapper = () => (
+    <Grid hasGutter>
+      <GridItem span={8}>
+        <Title size="2xl" headingLevel="h1">
+          Add credentials to this new device{" "}
+        </Title>
+        <br />
+      </GridItem>
+      <GridItem span={8}>
+        <AddCredential />
+      </GridItem>
+    </Grid>
+  );
+
+  const addCredentials = {
+    name: "Add credentials",
+    component: <AddCredentialWrapper />
+  };
+
+  const reviewForm = {
+    name: "Review",
+    component: <p>Review</p>
+  };
+
+  const onConnectionChange = (_: boolean, event: any) => {
+    const data = event.target.value;
+    if (data) {
+      setConnectionType(data);
+    }
+  };
+
+  const handleSave = async () => {
+    // if (name) {
+    //   const getVariables = () => {
+    // let variable: any = {
+    //   metadata: {
+    //     namespace: namespace
+    //   },
+    //   spec: {
+    //     type: addressType.toLowerCase(),
+    //     plan: plan,
+    //     address: addressName
+    //   }
+    // };
+    // if (addressType && addressType.trim().toLowerCase() === "subscription")
+    //   variable.spec.topic = topic;
+    // return variable;
+    // };
+    // const variables = {
+    // a: getVariables(),
+    // as: name
+    // };
+    // await setAddressQueryVariables(variables);
+    // }
+
+    onCloseDialog();
+  };
+
+  const steps = [
+    {
+      name: "Device information",
+      component: <DeviceInformation />
+    },
+    {
+      name: "Connection Type",
+      component: (
+        <ConnectionType
+          connectionType={connectionType}
+          onConnectionChange={onConnectionChange}
+        />
+      )
+    }
+  ];
+
+  if (connectionType) {
+    if (connectionType === "directly") {
+      steps.push(addCredentials);
+    } else {
+      steps.push(addGateway);
+    }
+    steps.push(reviewForm);
+  }
+
+  const handleNextIsEnabled = () => {
+    return true;
+  };
+
+  const CustomFooter = (
+    <WizardFooter>
+      <WizardContextConsumer>
+        {({ activeStep, onNext, onBack, onClose }) => {
+          if (
+            activeStep.name === "Device information" ||
+            activeStep.name === "Connection Type"
+          ) {
+            return (
+              <>
+                <Button
+                  variant="primary"
+                  type="submit"
+                  onClick={onNext}
+                  className={
```

Comment:
why are you not using disabled property for disable button ? you should use disabled .

Reply:
Currently I didn't handled the form validation to disable or enable next and Finish button.